From 69b32c90cbe432946d173718385a7301b0f04dff Mon Sep 17 00:00:00 2001
From: Honore Tricot <honorex.tricot@intel.com>
Date: Fri, 26 Oct 2012 09:29:02 +0200
Subject: [PORT FROM MAIN]wifi: hide password in logs

BZ: 64477

When the SET_NETWORK command is sent in order to set the password,
the password value appears in clear inside logs.

This fix catch the command SET_NETWORK + password attribute and remove
its value from string to send into log.

Change-Id: I67cb8ee1b1de5a5ff4ebd878f38f86c8d5bf63db
Orig-Change-Id: I15ebd0e5bfb45b211e9738ccbe908f3ddd23f605
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: yes
Signed-off-by: Paul Naoumenko <paulx.naoumenko@intel.com>
---
 wifi/wifi.c |   15 ++++++++++++++-
 1 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/wifi/wifi.c b/wifi/wifi.c
index 8a486db..4f681e9 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -647,7 +647,20 @@ int wifi_connect_to_supplicant()
 
 void log_cmd(const char *cmd)
 {
-    LOGI("CMD: %s\n", cmd);
+    if (strstr (cmd, "SET_NETWORK") && strstr(cmd, "password")) {
+        char *pbuf = malloc(strlen(cmd) + 1);
+        if (pbuf) {
+            strncpy(pbuf, cmd, strlen(cmd) + 1);
+            pbuf[strlen(cmd)]='\0';
+            char *p = strchr(pbuf, '\"');
+            if (p)
+                *p = '\0';
+            LOGI("CMD: %s\n", pbuf);
+        }
+        free(pbuf);
+    }
+    else
+        LOGI("CMD: %s\n", cmd);
 }
 
 void log_reply(char *reply, size_t *reply_len)
-- 
1.7.4.1

